tools/journal_proto_processor.cpp

```diff
@@ -606,24 +620,55 @@ void JournalProtoProcessor::ProcessInOut(
       // instead.
       // TODO(phl): This is probably incorrect for a non-pointer, but then we'll
       // notice it because it won't compile.
-      std::string const cxx_run_field_deserializer_getter =
-          Contains(out_, field_descriptor) ? "0" : cxx_run_field_getter;
-      cxx_run_body_prolog_[descriptor] +=
-          "  auto " + run_local_variable + " = " +
-          field_cxx_optional_pointer_fn_[field_descriptor](
-              ToLower(name) + ".has_" + field_descriptor_name + "()",
-              field_cxx_deserializer_fn_[field_descriptor](
-                  cxx_run_field_deserializer_getter)) +
-          ";\n";
+      if (Contains(out_, field_descriptor)) {
+        // For an out parameter, we need to allocate some uninitializeed
+        // storage.  The |make_unique| below requires that
+        // |field_cxx_type_[field_descriptor]| be default-constructible.
+        // For more generality, we could initialize the |unique_ptr| to
+        // |reinterpret_cast<T*>(std::operator new(sizeof(T)))|.
+        cxx_run_body_prolog_[descriptor] +=
+            "  auto const " + run_local_variable + " = std::make_unique<" +
+            field_cxx_direct_type_[field_descriptor] + ">();\n";
+        // Give a deterministic value, yet ensure that the function does not
+        // rely on the initial value of an out argument.
+        // Note that the resulting object may be invalid, so we must ensure
+        // that assigning to it isn't undefined behaviour; we thus require
+        // |is_trivially_copyable|, which means that copy and move assignment
+        // and construction are equivalent to |memmove|, and that destruction
+        // has no effect.
+        cxx_run_body_prolog_[descriptor] +=
+            "  std::memset(" + run_local_variable + ".get(), 0xAB, sizeof (" +
+            field_cxx_direct_type_[field_descriptor] + "));\n";
+        cxx_run_body_prolog_[descriptor] +=
+            "  static_assert(\n    std::is_trivially_copyable<" +
+            field_cxx_direct_type_[field_descriptor] +
+            ">::value,\n    \"out parameter |" +
+            run_local_variable + "| of |" +
+            field_descriptor->containing_type()->containing_type()->name() +
+            "| must have a trivially copyable type\");\n";
+      } else {
+        std::string const cxx_run_field_deserializer_getter =
+            cxx_run_field_getter;
```

Comment:
This variable is no longer needed, it was just here to isolate a `?:` expression.